lib/glific_web/controllers/api/v1/trial_account_controller.ex

```diff
@@ -0,0 +1,140 @@
+defmodule GlificWeb.API.V1.TrialAccountController do
+  @moduledoc """
+  Controller for allocating trial accounts to users via an API endpoint.
+  """
+  use GlificWeb, :controller
+
+  alias Glific.{
+    Contacts,
+    Contacts.Contact,
+    Partners.Organization,
+    Repo,
+    Users,
+    Users.User
+  }
+
+  alias GlificWeb.API.V1.RegistrationController
+
+  import Ecto.Query
+
+  @doc """
+  Allocates a trial account to a user.
+  """
+  @spec trial(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def trial(conn, params) do
+    token = get_req_header(conn, "x-api-key") |> List.first()
+    expected_token = get_token()
+
+    if token == expected_token do
+      phone = params["phone"]
+
+      with {:ok, _message} <-
+             RegistrationController.verify_otp(phone, params["otp"]),
+           {:ok, organization} <- get_available_trial_account(),
+           {:ok, contact} <- create_trial_contact(organization, phone, params),
+           {:ok, _user} <-
+             create_trial_user(organization, contact, phone, params) do
+        json(conn, %{
+          success: true,
+          data: %{
+            login_url: "https://#{organization.shortcode}.glific.com"
+          }
+        })
+      else
+        {:error, error_list} when is_list(error_list) ->
+          conn
+          |> json(%{
+            success: false,
+            error: "Invalid OTP"
+          })
+
+        {:error, :no_available_accounts} ->
+          json(conn, %{
+            success: false,
+            error: "No trial accounts available at the moment"
+          })
+
+        {:error, _} ->
+          conn
+          |> json(%{
```

Comment:
```suggestion
         |> put(500)
          |> json(%{
```

For this ,we should return HTTP 500